examples/console/main.go

```diff
@@ -103,6 +103,27 @@ func run() error {
 		FeeAccount:        accountKey,
 		FeeAccountSigners: []*keypair.Full{signerKey},
 	}
+	events := make(chan agent.Event)
```

Comment:
💡   thoughts on adding a "poll" command to the example similar to as you described in the description:
> They can simply use a buffered channel and poll when they want to.

I think logging doesn't show the  power of this events channel. But being able to poll on command the events could simulate what a user might do

wouldn't be in this PR though, just an idea that popped into my head

Reply:
going to approve since this would be in a separate PR if we decided to do

Reply:
We could have some example tests for sure that demonstrate different methods of use.